cart-detail/cart-detail.component.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+angular.module('cartDetail').
+component('cartDetail',{
+    templateUrl: "cart-detail/cart-detail.template.html",
+    controller: ['$http','$scope',function CartDetailController($http,$scope){
+
+        var self = this;
+        $http.get('http://localhost:3000/cart').then(function(response){
+            self.items = response.data;
+            var item;
+            self.totalPrice=0;
+            self.cartCount = self.items.length;
+            for(item in self.items){
+                self.totalPrice += self.items[item].price * self.items[item].count;  
+            }
+        });
+
+        $http.get('http://localhost:3000/products').then(function(response){

```

Comment:
This hard-coded string can be kept on any property file/constants because when lets suppose port gets changed we do not have to change every location.